python/src/aiconfig/scripts/aiconfig_cli.py

```diff
@@ -64,8 +65,33 @@ def _sigint(procs: list[subprocess.Popen[bytes]]) -> Result[str, str]:
         p.send_signal(signal.SIGINT)
     return Ok("Sent SIGINT to frontend servers.")
 
+def is_port_in_use(port: int) -> bool:
```

Comment:
I think this is fine for now, but there are both functional and UX problems with this approach.

Functional: It might be unlikely, but technically you can have a race condition where the port gets used after this returns but before the flask server starts.

UX: If I pass a port explicitly and it's already in use, I expect an error. I don't want the app to find the next open port for me. 

Here's what I would have done:

- If user provides port, don't do any special logic
- Otherwise, loop over all possible ports (e.g. starting at 8080) and try to start the server until one works. 


cc @saqadri , @rossdanlm

Reply:
Makes sense, but I think this could be seen as aan enhancement instead of a bug. I made a GH issue for this in https://github.com/lastmile-ai/aiconfig/issues/633 as a bootcamp task that others can do if they like